docs/spec/staking/transactions.md

```diff
@@ -38,275 +17,150 @@ type TxCreateValidator struct {
-
-    err := setInitialCommission(validator, tx.Commission, blockTime)
-    if err != nil return err // must be able to set initial commission correctly
-
-    // set the validator and public key
-    setValidator(validator)
-    setValidatorByPubKeyIndex(validator)
+This message is expected to fail if: 
 
-    // delegate coins from tx.DelegatorAddr to the validator
-    err := delegate(tx.DelegatorAddr, tx.Delegation, validator)
-    if err != nil return err // must be able to set delegation correctly
+ - another validator with this operator address is already registered
+ - another validator with this pubkey is already registered
+ - the initial self-delegation tokens are of a denom not specified as the
+   bonding denom 
+ - the commission parameters are faulty, namely:
+   - `MaxRate` is either > 1 or < 0 
+   - the initial `Rate` is either negative or > `MaxRate`
+   - the initial `MaxChangeRate` is either negative or > `MaxRate`
+ - the description fields are too large
+ 
+This message creates and stores the `Validator` object at appropriate
+indexes.  Additionally a self-delegation is made with the initial tokens
+delegation tokens `Delegation`.  the validator always starts as unbonded but
+may be bonded in the first end-block. 
 
-    tags := createTags(tx)
-    return tags
-```
 
-## TxEditValidator
+## MsgEditValidator
 
-If either the `Description`, `Commission`, or the `ValidatorAddr` need to be
-updated, the `TxEditCandidacy` transaction should be sent from the operator
-account:
+The `Description`, `CommissionRate` of a validator can be updated using the
+`MsgEditCandidacy`.  
 
 ```golang
-type TxEditCandidacy struct {
+type MsgEditCandidacy struct {
     Description     Description
     ValidatorAddr   sdk.ValAddress
     CommissionRate  sdk.Dec
 }
+```
 
-editCandidacy(tx TxEditCandidacy):
-    validator, ok := getValidator(tx.ValidatorAddr)
-    if !ok return err // validator must exist
-
-    // Attempt to update the validator's description. The description provided
-    // must be valid.
-    description, err := updateDescription(validator, tx.Description)
-    if err != nil return err
-
-    // a validator is not required to update it's commission rate
-    if tx.CommissionRate != nil {
-        // Attempt to update a validator's commission rate. The rate provided
-        // must be valid. It's rate can only be updated once a day.
-        err := updateValidatorCommission(validator, tx.CommissionRate)
-        if err != nil return err
-    }
-
-    // set the validator and public key
-    setValidator(validator)
+This message is expected to fail if: 
 
-    tags := createTags(tx)
-    return tags
-```
+ - the initial `CommissionRate` is either negative or > `MaxRate`
+ - the `CommissionRate` has already been updated within the previous 24 hours
+ - the `CommissionRate` is > `MaxChangeRate`
+ - the description fields are too large
 
-### TxDelegate
+This message stores the updated `Validator` object. 
```

Comment:
Maybe talk about the keeper functions used? More specificity here seems appropriate.

Reply:
added a link as per my previous comment